torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -660,71 +660,112 @@ def preprocessor(inputs, layout):
                 *normalize_shapes(*maybe_to_dense(*reorder_and_filter(inputs, layout)))
             )
 
-        def postprocessor(output):
-            if isinstance(output, ir.TensorBox):
-                # prepack the weight as input to the template buffer
-                template_buffer = ir.InputsKernel.unwrap_storage_for_input(output)
-                assert isinstance(template_buffer, ir.CppTemplateBuffer)
-                new_input_nodes, _ = reorder_and_filter(input_nodes, layout)
-
-                W_node = new_input_nodes[1]
-                assert W_node.get_name() in V.graph.constants
-                W = V.graph.constants[W_node.get_name()]
-                new_input_nodes[1] = W
-                new_input_nodes, _ = pack_weight(
-                    *normalize_shapes(*maybe_to_dense(new_input_nodes, layout))
+        def prune_tensors(input_nodes, new_input_nodes):
+            def share_storage(base_tensor: torch.Tensor, comp_tensor: torch.Tensor):
+                return (
+                    base_tensor.is_mkldnn == comp_tensor.is_mkldnn
+                    and base_tensor.dtype == comp_tensor.dtype
+                    and base_tensor.device == comp_tensor.device
+                    and (
+                        (
+                            not base_tensor.is_mkldnn
+                            and (
+                                base_tensor.untyped_storage().data_ptr()
+                                == comp_tensor.untyped_storage().data_ptr()
+                            )
+                        )
+                        or (
+                            base_tensor.is_mkldnn
+                            and (
+                                torch.ops.mkldnn.data_ptr(base_tensor)
+                                == torch.ops.mkldnn.data_ptr(comp_tensor)
+                            )
+                        )
+                    )
                 )
 
+            def get_candidates(input_nodes, new_input_nodes):
+                def node_share_storage(
+                    candidate_node: ir.IRNode, new_input_nodes: List[ir.IRNode]
+                ):
+                    for new_node in new_input_nodes:
+                        if (
+                            isinstance(new_node, (ir.TensorBox, ir.StorageBox))
+                            and new_node.get_name() in V.graph.constants
+                        ):
+                            candidate_tensor = V.graph.constants[
+                                candidate_node.get_name()
+                            ]
+                            comp_tensor = V.graph.constants[new_node.get_name()]
```

Comment:
To name things consistently?
```suggestion
                            new_tensor = V.graph.constants[new_node.get_name()]
```

Reply:
Thanks and changed.